### ISSUE TITLE
Getting NPE when I compare two Yaml or JSON file.

### ISSUE THREAD

**Issue by bmistry13** [OpenAPITools/openapi-diff#516]
Here is stack trace with issue:

```
docker run --rm -t -v /Users/bmistry/Downloads:/specs:ro openapitools/openapi-diff:latest /specs/old.yml /specs/new.yml
Unexpected exception. Reason: null
java.lang.NullPointerException
	at org.openapitools.openapidiff.core.compare.ParametersDiff.pathUnchangedParametersChanged(ParametersDiff.java:102)
	at org.openapitools.openapidiff.core.compare.ParametersDiff.diff(ParametersDiff.java:88)
	at org.openapitools.openapidiff.core.compare.OperationDiff.diff(OperationDiff.java:78)
	at org.openapitools.openapidiff.core.compare.PathDiff.diff(PathDiff.java:39)
	at org.openapitools.openapidiff.core.compare.PathsDiff.lambda$diff$3(PathsDiff.java:90)
	at java.util.LinkedHashMap$LinkedKeySet.forEach(LinkedHashMap.java:559)
	at org.openapitools.openapidiff.core.compare.PathsDiff.diff(PathsDiff.java:47)
	at org.openapitools.openapidiff.core.compare.OpenApiDiff.compare(OpenApiDiff.java:96)
	at org.openapitools.openapidiff.core.compare.OpenApiDiff.compare(OpenApiDiff.java:64)
	at org.openapitools.openapidiff.core.OpenApiCompare.fromSpecifications(OpenApiCompare.java:102)
	at org.openapitools.openapidiff.core.OpenApiCompare.fromLocations(OpenApiCompare.java:91)
	at org.openapitools.openapidiff.cli.Main.main(Main.java:175)
```

**Comment by joschi** [comment]
@bmistry13 Please provide the `old.yml` and `new.yml` OpenAPI specs or alternatively a minimal reproducible example for this issue.

**Comment by cliffred** [comment]
Got the same error with same stack trace. The parameters in the DiffContext is empty.

The minimal example to reproduce is to simple rename a parameter:

```yaml
      parameters:
        - name: limit #rename this in the new yaml
          in: query
          description: How many items to return at one time (max 100)
          required: false
          schema:
            type: integer
            maximum: 100
            format: int32
```
 
[specs.zip](https://github.com/OpenAPITools/openapi-diff/files/11743532/specs.zip)

**Comment by cliffred** [comment]
This issue does not occur in v2.0.1.
I got this while using 2.1.0-beta.5.

**Comment by stoeren** [comment]
This issue has been resolved in version `2.1.0-beta7` and is a duplicate of #488
